manager/client/acquire/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"os"
+	"time"
+	"log"
+	"context"
+	"google.golang.org/grpc"
+	"strconv"
+	rpc_license "github.com/enfabrica/enkit/manager/rpc"
+)
+
+func main() {
```

Comment:
As mentioned over chat, I'm very worried about having an `acquire` command and a separate `release` command. Whenever the build is killed, or fails, or there's a coding error in a rule, .. if we fail to call `release`, we leak licenses, I fear this won't work reliably.

My ideal would be to have a command `with-license` that wraps another command, something like:

```
./with-license vendor feature feature -- vivado -mode tcl ...
```

.. eg, runs the command after "--" passing all the remaining flags to it, after grabbing the licenses. When the command exits, the licenses are automatically freed. No need to call a separate release command.

If you implement the suggested changes in the `.proto` file, we end up with:
- having to call `Acquire` until it succeeds, giving us a `ticket`.
- having to call `KeepaliveRequest` (aka `RenewLease` if you accept the suggestion) every 30 seconds.
- having to call `Release` with the ticket (at the end of the process).

To implement something like I described, the code could look something like:

```
func main() {
  // ... flag parsing ...
  vendor := ...
  features := []string{...}
  command := []string{... leftover argv parsing ...}
  
  // ... grpc initialization ...

  // Try to grab a license.
  // Use import "github.com/lib/retry"
  var response rpc_license.AcquireResponse
  if err := retry.New(retry.WithWait(60), retry.WithAttempts(30)).Run(func () error {
    response, err = client.Acquire(context.Background(), &rpc_license.AcquireRequest{Vendor: vendor, Feature: features, Quantity: int32(quantity)})
    return err  
  }); err != nil {
    log.Fatalf("could not acquire license - exiting")
  }

  // If we got here, it means we successfully got a license. Retrying succeeded.
  // Now: 0) make sure we release the license as quickly as possible once done,
  // 1) keep the license alive, and 2) run the command.

  // defer causes the function to invoke at return, no matter what.
  defer Release(client, response.Ticket)
  // go starts a separate thread.
  go Refresh(client, response.Ticket, 60 * time.Second)  

  // import "os/exec"
  cmd := exec.Command(command[0], command[1:]...)
  cmd.Stdin = os.Stdin
  cmd.Stdout = os.Stdout

  // Finally run the command.
  if err := cmd.Run(); err != nil {
    if exitError, ok := err.(*exec.ExitError); ok {
        os.Exit(exitError.ExitCode())
    }
    // Running the command failed with an unknown exit code. log something.
    os.Exit(255)
  }
  return // this will exit 0.
}
```

Release and Refresh could be implemented as something like:
```
func Refresh(client *rpc_license.LicenseClient, ticket string, delay time.Duration) {
  for {
    if err := client.RenewLease(context.Background(), &rpc_license.KeepaliveRequest{ticket: ticket}); err != nil {
      time.Sleep(5 * time.Second)  // we failed to renew the ticket, retry faster to avoid expiry!
      // TODO: stop the loop if it's a fatal error. Eg, maybe the server restarted, and does not know about the ticket!
    } else {
      time.Sleep(dealy / 2) // renew the ticket well before it expires.
    }
  }
}

func Release(client, ticket) {
  client.Release(...) // no real need to handle errors here. Worst case, the lease will expire and the license freed.
}
```